pkg/asset/machines/gcp/machines.go

```diff
@@ -96,7 +97,8 @@ func provider(clusterID string, platform *gcp.Platform, mpool *gcp.MachinePool,
 	}
 
 	instanceServiceAccount := fmt.Sprintf("%s-%s@%s.iam.gserviceaccount.com", clusterID, role[0:1], platform.ProjectID)
-	if credentialsMode == types.PassthroughCredentialsMode {
+	// Passthrough service accounts are only needed for GCP XPN.
+	if len(platform.NetworkProjectID) > 0 && credentialsMode == types.PassthroughCredentialsMode {
```

Comment:
The XPN enhancement (currently) also allows manual mode. Do we need to cover that here ?